docs/resources/shadow.md.erb

```diff
@@ -5,148 +5,255 @@ platform: linux
 * The minimum number of days a password must exist, before it may be changed
 * The maximum number of days after which a password must be changed
 * The number of days a user is warned about an expiring password
 * The number of days a user must be inactive before the user account is disabled
-* The number of days a user account has been disabled
+* The date on which a user account was disabled, as the number of days since Jan 1 1970
 
 These entries are defined as a colon-delimited row in the file, one row per user:
 
     dannos:Gb7crrO5CDF.:10063:0:99999:7:::
 
+The `shadow` resource understands this format, allows you to search on the fields, and exposes the selected users' properties.
+
 <br>
 
-## Syntax
+## Resource Parameters
 
-A `shadow` resource block declares user properties to be tested:
+The `shadow` resource takes one optional parameter: the path to the shadow file.  If omitted, `/etc/shadow` is assumed.
 
+    # Expect a file to exist at the default location and have 32 users
     describe shadow do
-      its('users') { should_not include 'forbidden_user' }
+      its('count') { should eq 32 }
     end
 
-Properties can be used as a single query:
+    # Use a custom location
+    describe shadow('/etc/my-custom-place/shadow') do
+      its('count') { should eq 32 }
+    end
 
-    describe shadow.user('root') do
+## Examples
+
+A `shadow` resource block uses `where` to filter entries from the shadow file. If `where` is omitted, all entries are selected.
+
+    # Select all users. Among them, there should not be a user with the name 'forbidden_user'.
+    describe shadow do
+      its('users') { should_not include 'forbidden_user' }
+    end
+
+    # Ensure there is only one user named 'root' (Select all with name 'root', then count them).
+    describe shadow.where(user: 'root') do
       its('count') { should eq 1 }
     end
 
-Use the `.where` method to find properties that match a value:
+You can use `where` to match any of the [filter criteria](#filter_criteria) supported. `where` has a method form for simple equality, and a block form for more complex queries.
 
-    describe shadow.where { min_days == '0' } do
-      its ('users') { should include 'nfs' }
+    # Method form, simple
+    # Select just the root user (direct equality)
+    describe shadow.where(user: 'root') do
+      its ('count') { should eq 1 }
     end
 
-    describe shadow.where { password =~ /[x|!|*]/ } do
-      its('count') { should eq 0 }
+    # Method form, with a regex
+    # Select all users whose names begin with smb
+    describe shadow.where(user: /^smb/) do
+      its ('count') { should eq 2 }
     end
 
-The following properties are available:
+    # Block form
+    # Select users whose passwords have expired
+    describe shadow.where { expiry_date > 0 } do
+      # This is more clear a a test
+      its('count') { should eq 0 }
+      # But if the count test fails, this test gives clearer diagnostics
```

Comment:
Suggest `# But if the count test fails, this test will output the users that are causing the failure`